lib/kitto/hooks.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Kitto.Hooks do
+  @moduledoc """
+  Kitto Hooks enable an alternative to Jobs for feeding data into dashboards.
+  Hooks enable remote services to push data into Kitto using webhooks.
+
+  Just like jobs, hooks are loaded at runtime from the `hooks/` directory at
+  the root of the application. Hooks can be defined as follows:
+
+      use Kitto.Hooks.DSL
+      hook :hello do
+        {:ok, body, _} = read_body conn
+        broadcast! :hello, body |> Poison.decode!
+      end
+  """
+
+  use GenServer
+
+  def start_link do
+    GenServer.start_link(__MODULE__, :ok, name: :hook_registry)
+  end
+
+  def init(:ok) do
+    load_hooks
+    {:ok, %{}}
+  end
+
+  @doc """
+  Lookups a hook from the registry
+  """
+  def lookup(hook) do
+    GenServer.call(:hook_registry, {:lookup, hook})
+  end
+
+  @doc """
+  Registers a new hook into the registry
+  """
+  def register(name, block) do
```

Comment:
This should be like https://github.com/kittoframework/kitto/blob/master/lib/kitto/runner.ex#L31

where the first param is a reference to the process.